operator/v1/types_ingress.go

```diff
@@ -225,6 +233,97 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
+	// Profile is one of three possible security configurations from [1]:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// If unset, the default is "intermediate".
+	//
+	// +optional
+	Profile *SecurityProfileType `json:"profile,omitempty"`
+	// CustomSettings defines the schema for customizing the security settings of
+	// IngressController connections.
+	//
+	// +optional
+	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// SecurityProfileOldType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "old"
+	// SecurityProfileIntermediateType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "intermediate"
+	// SecurityProfileModernType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "modern"
```

Comment:
Some legacy definitions use all lower-case, but we've been using initial upper-case names for string constants in new APIs, so these should probably be changed: "old"→"Old", "intermediate"→"Intermediate", and "modern"→"Modern".

Reply:
Will do.